### ISSUE TITLE
Publish to DefinitelyTyped as @types/rmwc

### ISSUE THREAD

**Issue by kctang** [ncknt/rmwc-typed#5]
Thanks for providing `rmwc-typed`. Can we publish it to DefinitelyTyped as `@types/rmwc`?

**Comment by ncknt** [comment]
We're discussing whether to publish to `@types/rmwc` or just have the definition files in the project: https://github.com/jamesmfriedman/rmwc/issues/40#issuecomment-355770067

@kctang: Feel free to comment there too!

**Comment by SleeplessByte** [comment]
This project as is should not be published to DefinitelyTyped. I suggest to mimic the flow types much closer, so that edits can be made by anyone. 

For example `/Base/index.d.ts` would be something like
```TypeScript
declare module 'rmwc/Base' {
  export { simpleTag, SimpleTagProps } from 'rmwc/Base/simpleTag'
  export { withRipple, WithRippleProps } from 'rmwc/Base/withRipple'
  export { withTheme, WithThemeProps } from 'rmwc/Base/withTheme'
  export { withMDC } from 'rmwc/Base/withMDC'
  export { withMDCToggle } from 'rmwc/Base/withMDCToggle'
  export { noop } from 'rmwc/Base/noop'
}
```

Which is exactly the same exports the actual javascript library has (in flow + js).

**Comment by mutsys** [comment]
I agree with @SleeplessByte that it should not be published as is, but for different reasons.

1) As the typings are current written, they can only be imported into your project from the `index.d.ts` definition as:

```typescript
import {
    Toolbar
} from "rmwc";
```

This has the unfortunate side effect of importing all of the `rmwc` widgets into your project, not just the ones you selected for use. The import statement should be able to written instead as:

```typescript
import {
    Toolbar
} from "rmwc/Toolbar";
```

which is in accordance with the recommendations of the `rmwc` author, which will result in only the Toolbar widgets getting imported. Adjusting the type definitions to allow for this in no way prevents you from also having a "master" type definition file that imports everything.

2. The definition for Base should not be a plain `.ts` file, it should be a type definition file as well, i.e. - `Base.d.ts`.

**Comment by SleeplessByte** [comment]
If you apply what I said you solve both issues you describe; and I completely agree with those two points.

I think James (rmwc) is still looking to compile definitions automagically, which would be great.

**Comment by ncknt** [comment]
Thank you @mutsys for your PR. It's now been merged.
